### PR TITLE
Fix search by issue type

### LINKED ISSUE #1886: Issues sort by Type menu does not work
- Gitea version (or commit ref): 1.1.1+2-g2854c8a
- Git version: 2.12.2
- Operating system: Windows Server 2012
- Database (use `[x]`):
  - [x] MSSQL
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes
- Log gist: Not relevant

In the issues list, the 'Type' sorting menu is broken. None of the options affect the sorting or searching of the issues.

### PR DESCRIPTION
**Pull request by ethantkoenig**
Fixes #1886.

Other fixes/changes:
- Only display the "Type" menu when the user is logged-in (otherwise a search for "Assigned to you" doesn't make much sense)
- Fix ambiguous column references in `GetIssueStats(..)`
- Proper error handling for `GetIssueStats(..)`

### CODE REVIEW

File: models/issue.go [hunk 1]
```diff
@@ -1279,40 +1278,15 @@ func GetIssueStats(opts *IssueStatsOptions) (*IssueStats, error) {
...
-		stats.OpenCount, err = countSession(opts).
-			And("is_closed = ?", false).
-			Count(new(Issue))
-
-		stats.ClosedCount, err = countSession(opts).
-			And("is_closed = ?", true).
-			Count(new(Issue))
-	case FilterModeCreate:
-		stats.OpenCount, err = countSession(opts).
-			And("poster_id = ?", opts.PosterID).
```
Comment (ethantkoenig) [inline models/issue.go:1293]: These conditions are already handled in `countSession(..)`, so we don't need to repeat them again here.

File: templates/repo/issue/list.tmpl [hunk 2]
```diff
@@ -68,19 +68,21 @@
...
+				{{if .IsSigned}}
+					<!-- Type -->
+					<div class="ui dropdown type jump item">
+						<span class="text">
+							{{.i18n.Tr "repo.issues.filter_type"}}
+							<i class="dropdown icon"></i>
+						</span>
+						<div class="menu">
+							<a class="{{if eq .ViewType "all"}}active{{end}} item" href="{{$.Link}}?q={{$.Keyword}}&type=all&sort={{$.SortType}}&state={{$.State}}&labels={{.SelectLabels}}&milestone={{$.MilestoneID}}&assignee={{$.AssigneeID}}">{{.i18n.Tr "repo.issues.filter_type.all_issues"}}</a>
+							<a class="{{if eq .ViewType "assigned"}}active{{end}} item" href="{{$.Link}}?q={{$.Keyword}}&type=assigned&sort={{$.SortType}}&state={{$.State}}&labels={{.SelectLabels}}&milestone={{$.MilestoneID}}&assignee={{.SignedUser.ID}}">{{.i18n.Tr "repo.issues.filter_type.assigned_to_you"}}</a>
```
Comment (ethantkoenig) [inline templates/repo/issue/list.tmpl:80]: `assignee` param updated to be `.Signed.User.ID` (might be hard to tell what the non-whitespace changes are)

**Comment by lunny** [comment]
Maybe some integration tests.

**Comment by ethantkoenig** [comment]
@lunny Added an integration test

**Comment by lafriks** [comment]
It fails on postgresql with error `pq: operator does not exist: boolean = integer`

**Comment by ethantkoenig** [comment]
@lafriks Fixed

**Comment by lunny** [comment]
LGTM

**Comment by bkcsoft** [comment]
LGTM

**Comment by bkcsoft** [comment]
ffs LG-TM...

**Pull request merged**